docs: add source code build documentation (#4210)
web-infra-dev/modern.js: packages/document/builder-doc/docs/en/config/experiments/sourceBuild.md

- **Type:** `boolean`

> 

Reviewer: remove empty line

- **Default:** `false`
- **Bundler:** `only support webpack`

> Used to enable the ability to source-code build. When this option is turned on, Builder reads the source file corresponding to the `source` field of the subproject package.json and compiles it.

Reviewer: ```suggestion
Used to enable the ability for source code building. When this configuration option is enabled, Builder will read the source code files corresponding to the `source` field in the sub-project's package.json and compile them.
```

- **Type:** `boolean`
- **Default:** `false`
- **Bundler:** `only support webpack`

Used to enable the ability for source code building. When this configuration option is enabled, Builder will read the source code files corresponding to the `source` field in the sub-project's package.json and compile them.

```ts
export default {
  experiments: {
    sourceBuild: true,
  },
};
```

> More detail can see [[Advanced-Source-code Build Mode]](en/guide/advanced/source-build.md)。

Reviewer: ```suggestion
More detail can see ["Advanced - Source-code Build Mode"](en/guide/advanced/source-build.md)。
```
